frontend/js/components/ecosystems/LocationForm.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import Hr from './../atoms/hr';
+import Alert from './../organisms/Alert';
+import TextField from './../organisms/TextField';
+
+import {
+    getLocationByGPS,
+    getLocationByAddress
+} from './../../redux/actions/pick-survey-actions';
+
+class LocationForm extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {address: ''};
+    }
+
+    updateAddress(event) {
+        const address = event.target.value;
+        this.setState({
+            address: address
+        });
+    }
+
+    search(event) {
+        event && event.preventDefault();
+        this.props.dispatch(getLocationByAddress(this.state.address), () => {
+            this.setState({address: ''});
+        });
+    }
+
+    geolocate() {
+        this.props.dispatch(getLocationByGPS());
+    }
+
+    render() {
+        return (
+            <form className="location-form" onSubmit={this.search.bind(this)}>
+                <section>
+                    {this.props.status.message &&
+                        <Alert
+                            level={this.props.status.level}
+                            message={this.props.status.message} />
+                    }
+                </section>
+                <section className="by-address">
+                    <TextField
+                        label="Locate Using Address"
+                        name="address"
+                        value={this.state.value}
```

Comment:
This works (because JavaScript), but it we should be setting `this.state.address` instead.